src/hooks/useConfigure.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect } from 'react';
+import { Alert, useColorScheme } from 'react-native';
+import { SUBSCRIBER_EVENTS } from '@walletconnect/core';
+import { ExplorerCtrl } from '../controllers/ExplorerCtrl';
+import { OptionsCtrl } from '../controllers/OptionsCtrl';
+import { ConfigCtrl } from '../controllers/ConfigCtrl';
+import { ClientCtrl } from '../controllers/ClientCtrl';
+import { AccountCtrl } from '../controllers/AccountCtrl';
+import { WcConnectionCtrl } from '../controllers/WcConnectionCtrl';
+import type { ProviderMetadata } from '../types/coreTypes';
+import { createUniversalProvider } from '../utils/ProviderUtil';
+
+interface Props {
+  projectId: string;
+  providerMetadata: ProviderMetadata;
+  relayUrl?: string;
+}
+
+export function useConfigure({ projectId, relayUrl, providerMetadata }: Props) {
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const resetApp = useCallback(() => {
+    ClientCtrl.resetSession();
+    AccountCtrl.resetAccount();
+    WcConnectionCtrl.resetConnection();
+  }, []);
+
+  const onSessionDelete = useCallback(
+    ({ topic }: { topic: string }) => {
+      const sessionTopic = ClientCtrl.sessionTopic();
+      if (topic === sessionTopic) {
+        resetApp();
+      }
+    },
+    [resetApp]
+  );
+
+  const onDisplayUri = useCallback(async (uri: string) => {
+    WcConnectionCtrl.setPairingUri(uri);
+  }, []);
+
+  useEffect(() => {
+    ConfigCtrl.setThemeMode(isDarkMode ? 'dark' : 'light');
+  }, [isDarkMode]);
+
+  /**
+   * Fetch wallet list
+   */
+  useEffect(() => {
+    async function fetchWallets() {
+      try {
+        if (!ExplorerCtrl.state.wallets.total) {
+          await ExplorerCtrl.getMobileWallets({ version: 2 });
+          OptionsCtrl.setIsDataLoaded(true);
+        }
+      } catch (error) {
+        Alert.alert('Error', 'Error fetching wallets');
+      }
+    }
+
+    ConfigCtrl.setConfig({ projectId });
+    fetchWallets();
```

Comment:
Wallet fetching should prob be in a separate effect, as we don't want to re-fetch when project id changes (unless you have caching going already)